lib/container.js

```diff
@@ -112,12 +112,28 @@ export default class Container {
     return new EventHandle(this.ee, USER_CHANGED, listener);
   }
 
-  signupWithUsername(username, password) {
-    return this._signup(username, null, password);
+  signupWithUsername(username, password, profile = null) {
+    const promise = this._signup(username, null, password);
+    if (profile === null) {
+      return promise;
+    }
+    return promise.then((user)=>
+      this._createProfile(user, profile).then(createdProfile =>
+        [user, createdProfile]
+      )
+    );
   }
 
-  signupWithEmail(email, password) {
-    return this._signup(null, email, password);
+  signupWithEmail(email, password, profile = null) {
+    const promise = this._signup(null, email, password);
+    if (profile === null) {
+      return promise;
+    }
+    return promise.then((user)=>
+      this._createProfile(user, profile).then(createdProfile =>
+        [user, createdProfile]
```

Comment:
Different return value type seem not a good idea. Agree?

`signupWithEmail(email, password)` will resolve to an object, while `signupWithEmail(email, password, {})` will resolved to an array of object.

Since we want an enchantment, not a breaking change here. Probably we don't want break existing code using `signupWithEmail`.

So a sensible solution will be 
`signupWithEmailAndProfile(email, password, {})` will resolved to an array of object.